Objects/codeobject.c

```diff
@@ -217,37 +218,40 @@ PyCode_NewWithPosOnlyArgs(int argcount, int posonlyargcount, int kwonlyargcount,
     }
 
     /* Check for any inner or outer closure references */
-    n_cellvars = PyTuple_GET_SIZE(cellvars);
-    if (!n_cellvars && !PyTuple_GET_SIZE(freevars)) {
+    assert(PyTuple_GET_SIZE(cellvars) < INT_MAX);
+    ncellvars = (int)PyTuple_GET_SIZE(cellvars);
+    assert(PyTuple_GET_SIZE(freevars) < INT_MAX);
+    nfreevars = (int)PyTuple_GET_SIZE(freevars);
+    if (!ncellvars && !nfreevars) {
         flags |= CO_NOFREE;
     } else {
         flags &= ~CO_NOFREE;
     }
 
-    n_varnames = PyTuple_GET_SIZE(varnames);
-    if (argcount <= n_varnames && kwonlyargcount <= n_varnames) {
+    assert(nlocals == PyTuple_GET_SIZE(varnames));
```

Comment:
This isn't guaranteed to be true, as the tests show.
I'm not sure what it means for `nlocals` to be different from `len(varnames)`.
Could we just ignore `nlocals` and set `co->co_nlocals = len(varnames)`? I don't any reason why not.

Reply:
Alternatively, and this might be better, raise an exception if they differ in calls to `CodeType()` or `co.replace()`

Reply:
They are different only if someone purposefully provides a different value through one of the legacy APIs.  The test failed because I added a check for the constraint.  I had fixed the test in my old PR and forgot to do so in this one.